### PR TITLE
feat: flatpak no escape, ide-wrapper, solves #6

### PR DESCRIPTION
**Pull request by appelgriebsch**

### CODE REVIEW

File: dev.zed.Zed.yaml
Comment (tomaswarynyca) [inline dev.zed.Zed.yaml:58]:
```suggestion
  - name: flatpak-wrapper
```

**Comment by jerome-benoit** [comment]
The build behaves as intended with env variables. 

@baronleonardo: do you also confirm?

**Comment by baronleonardo** [comment]
- this build `flatpak install --user https://dl.flathub.org/build-repo/124991/dev.zed.Zed.flatpakref` gave me 404
- for the current release, no it doesn't work

**Comment by jerome-benoit** [comment]
> * this build `flatpak install --user https://dl.flathub.org/build-repo/124991/dev.zed.Zed.flatpakref` gave me 404

And with the latest build?

**Comment by baronleonardo** [comment]
I can confirm that this build `flatpak install --user https://dl.flathub.org/build-repo/126729/dev.zed.Zed.flatpakref` is working as expected

**Comment by jerome-benoit** [comment]
> I can confirm that this build `flatpak install --user https://dl.flathub.org/build-repo/126729/dev.zed.Zed.flatpakref` is working as expected

@ShobuPrime: can it be considered for merge? It will permit people using the flatpak and nvidia driver to run Zed: https://github.com/zed-industries/zed/issues/17129#issuecomment-2323974189

**Review by tomaswarynyca — APPROVED** [review]

**Comment by jerome-benoit** [comment]
@appelgriebsch: mixing version upgrade + feature/fix in the same PR with not help at merging it ...

**Comment by appelgriebsch** [comment]
Not sure if it makes it easier to just merge this one and then update to latest version.... but feel free to decide what fit you better

**Comment by jerome-benoit** [comment]
> Not sure if it makes it easier to just merge this one and then update to latest version.... but feel free to decide what fit you better

The Zed flatpak repo is maintained by only one person? There's no backup?

**Comment by tomaswarynyca** [comment]
@ShobuPrime can you check this please?

**Comment by tomaswarynyca** [comment]
@appelgriebsch can you fix the PR so it doesn't delete 0.151.1?

**Comment by jerome-benoit** [comment]
> appelgriebsch force-pushed the feature/flatpak-no-escape branch from 46a2ecc to c72baa7. 30 minutes ago

By untangling the Zed version update from the wrapper introduction PR,  you do not need to maintain the later: no more conflicts to handle and the repo maintainer just has to eventually sync with main branch before merge.

**Comment by jerome-benoit** [comment]
> Very early in the building process @ConradIrwin has [mentioned](https://github.com/flathub/flathub/pull/5253#issuecomment-2119050116):
> 
> > It makes me very unhappy to think that we’d ship a partially non-functional zed on flathub
> 

The current Zed on flathub is not offering the same user experience than installing Zed via the install script: `zed` in not available via the CLI. Aligning the user experience with the behavior end users expect is actually a bug fix. 
Dunno why it was shipped without CLI usage consideration.  

@appelgriebsch: @ShobuPrime is right, changing ZED_FLATPAK_NO_ESCAPE current default is not necessary to properly expose `zed` binary through the CLI respecting env variables. It's in fact an orthogonal change that can be part of other PR and discussion more related to Zed flatpak security.

> I don't have a device with an Nvidia GPU at the moment, so I'm unable to test the Nvidia driver mentioned [here](https://github.com/flathub/dev.zed.Zed/pull/8#issuecomment-2324147894), but I trust some of you have already made sure it works.

As long as the `zed` binary has read access to the env variables, it will work.

**Comment by baronleonardo** [comment]
why had this been closed and deleted ?!

**Pull request closed without merging**
